README.md

```diff
@@ -72,9 +72,12 @@ will want to generate `x_indices` programmatically.
 - *test_size*: Fraction of data to use for test set in each iteration
 - *clf_info*: List of scikit-learn classifiers to use.
 - *permute*: List of booleans to indicate whether to generate a null model or not
-- *noshap*: Boolean indicating whether shap values are evaluated
+- *gen_shap*: Boolean indicating whether shap values are evaluated
 - *nsamples*: Number of samples to use for shap estimation
 - *l1_reg*: Type of regularizer to use for shap estimation
+- *plot_shap*: Boolean indicating whether shap values are plotted (csv will be created if gen_shap is true)
+- *plot_top_n_shap*: Number of top shap values to plot ranked by mean across splits
+- *confusion_matrix*: Boolean indicating whether to also compute SHAP values for TP, TN, FP, and FN
```

Comment:
i would suggest removing these. plots should always happen for the report if `gen_shap` is true. and output based shap figures would depend on whether the model is a classifier or regressor, but again should be generated whenever `gen_shap` is true. so please remove these options from the spec.

Reply:
Sure. Regarding `plot_top_n_shap` imagine there 700 features. How many should we plot? Top 16? Should we give them control over that? The full csv files will be outputted so people can make different size heatmaps of their choosing.

Reply:
ah that i can see as somewhat of an option. let's keep that for now. i can see how that may need to be adjusted. 

perhaps allow two interpretations of the value of this option: a fraction (0, 1) not including 0 (raise error) or 1+ (1 or more features)

so it would translate to `features_to_show = max(1, int(np.round(value*num_features))) if value < 1 else value`

Reply:
Sure. If people want to plot all features

Option A) Set variable to 1.0 (float) and 1 for top first feature:
```
	if plot_top_n_shap != 1.0:
		if plot_top_n_shap < 1:
			plot_top_n_shap = int(np.round(plot_top_n_shap * num_features))
		summary = summary.iloc[:plot_top_n_shap , :]
		filename += f"_top_{plot_top_n_shap}"
```
(if `plot_top_n_shap == 1.0` won't slice and will plot everything)


README:
```
- *plot_top_n_shap*: Number or proportion of top SHAP values to plot (e.g., 16 or 0.1 for top 10%). Set to 1.0 (float) to plot all features or 1 to plot top first feature.
```

Reply:
i would use 1 to just plot the top feature. it's easier to use the total number of number of features if you want to plot all (you already needed this to create the input index).

Reply:
we could allow distinguishing float 1.0 vs integer 1. but let's keep semantic consistency.